### PR TITLE
Add  console span exporter

### PR DESCRIPTION
**Pull request by plantfansam**
This PR

- Adds a console exporter to make debugging easier
- Moves the generation of span export data into its own set of methods, so that logic can be shared among different exporters

### CODE REVIEW

File: lib/opentelemetry/trace/recording_span.lua [hunk 1]
```diff
@@ -135,9 +135,53 @@ function _M.tracer_provider(self)
...
+    ret.parent_span_id = hex2bytes(ret.parent_span_id)
+    return ret
+end
+
+function _M.for_console_export(self)
+    local ret = "\n---------------------------------------------------------\n"
+    ret = ret .. util.table_as_string(self:for_export(), 2)
+    ret = ret .. "---------------------------------------------------------\n"
+    return ret
+end
```
Comment (yangxikun) [inline lib/opentelemetry/trace/recording_span.lua:180]: `for_otlp_export` and `for_console_export` is about export span logic, I think we should encapsulate in `lib/opentelemetry/trace/exporter/encoder.lua`. Then `encoder.to_oltp(span)` for oltp exporter, `encoder.to_console(span)` for console exporter.

File: lib/opentelemetry/trace/exporter/console_span_exporter.lua [hunk 2]
```diff
@@ -0,0 +1,43 @@
+-------------------------------------------------------------------------------
```
Comment (yangxikun) [inline lib/opentelemetry/trace/exporter/console_span_exporter.lua:1]: We can simplify the filename `console_span_exporter.lua` to `console.lua`.

**Review by yangxikun — CHANGES_REQUESTED** [review]

File: lib/opentelemetry/trace/exporter/console_span_exporter.lua (hunk 2 above)
Comment (plantfansam) [reply lib/opentelemetry/trace/exporter/console_span_exporter.lua]: 👍 f3d5607deeccd910eb5acd4864735d9d24042fd3

File: lib/opentelemetry/trace/recording_span.lua (hunk 1 above)
Comment (plantfansam) [reply lib/opentelemetry/trace/recording_span.lua:180]: Great idea - will do.

Comment (plantfansam) [reply lib/opentelemetry/trace/recording_span.lua]: 326750a75994b9c7b44f10dd0e573160d9b49022

File: lib/opentelemetry/trace/exporter/encoder.lua [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+-- @return String representation of span.
+--------------------------------------------------------------------------------
+function _M.for_console(span)
+    local ret = "\n---------------------------------------------------------\n"
+    ret = ret .. util.table_as_string(_M.for_export(span), 2)
+    ret = ret .. "---------------------------------------------------------\n"
+    return ret
+end
+
+function _M.to_otlp(span)
```
Comment (yangxikun) [inline lib/opentelemetry/trace/exporter/encoder.lua:85]: redundant code?

File: lib/opentelemetry/trace/recording_span.lua [hunk 4]
```diff
@@ -1,8 +1,8 @@
+local attribute   = require("opentelemetry.attribute")
+local event_new   = require("opentelemetry.trace.event").new
 local span_kind   = require("opentelemetry.trace.span_kind")
 local span_status = require("opentelemetry.trace.span_status")
-local event_new   = require("opentelemetry.trace.event").new
-local attribute   = require("opentelemetry.attribute")
-local util        = require("opentelemetry.util")
+local util    = require("opentelemetry.util")
```
Comment (yangxikun) [inline lib/opentelemetry/trace/recording_span.lua:5]: align indent

File: lib/opentelemetry/trace/exporter/console.lua [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+-- @return New console span exporter
+--------------------------------------------------------------------------------
+function _M.new()
+    return setmetatable({}, mt)
+end
+
+function _M.export_spans(self, spans)
+    local span_string = ""
+    for _, span in ipairs(spans) do
+        span_string = span_string .. encoder.to_console(span) .. "\n"
```
Comment (yangxikun) [inline lib/opentelemetry/trace/exporter/console.lua:27]: to_console? Or for_console?

**Review by yangxikun — CHANGES_REQUESTED** [review]

File: lib/opentelemetry/trace/exporter/console.lua (hunk 5 above)
Comment (plantfansam) [reply lib/opentelemetry/trace/exporter/console.lua]: 🤦 891ffb0ea474d45de11fed67ad5e7b578508b9c0

File: lib/opentelemetry/trace/exporter/encoder.lua (hunk 3 above)
Comment (plantfansam) [reply lib/opentelemetry/trace/exporter/encoder.lua]: It's Monday morning 🤦  f918d52fafb7d5aa1df1fa6b560c70b74c5674d1

File: lib/opentelemetry/trace/recording_span.lua (hunk 4 above)
Comment (plantfansam) [reply lib/opentelemetry/trace/recording_span.lua]: 770606bb8f8b6e99c0081e442c16d39714f03904

**Pull request merged**